src/main/java/it/pagopa/pn/service/desk/config/HttpConnector.java

```diff
@@ -0,0 +1,48 @@
+package it.pagopa.pn.service.desk.config;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.pdfbox.pdmodel.PDDocument;
+import org.springframework.http.MediaType;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+@Slf4j
+public class HttpConnector {
+
+    private HttpConnector(){
+        throw new IllegalCallerException("the constructor must not called");
+    }
+
+    public static Mono<PDDocument> downloadFile(String url) {
+        log.info("Url to download: {}", url);
+        try {
+            return WebClient
```

Comment:
Utilizziamo la funziona di WebClient di effettuare il download di file in chucnk per non appesantire la richiesta HTTP:

una cosa del genere (https://www.baeldung.com/webclient-stream-large-byte-array-to-file#webclient-for-any-file-size-with-databuffer):

Flux<DataBuffer> dataBufferFlux = WebClient.create()
.get()
.uri(url)
.accept(MediaType.APPLICATION_PDF)
.retrieve()
.bodyToFlux(DataBuffer.class);

        return DataBufferUtils.join(dataBufferFlux)
                .map(dataBuffer -> {
                    byte[] bytes = new byte[dataBuffer.readableByteCount()];
                    dataBuffer.read(bytes);
                    DataBufferUtils.release(dataBuffer);
                    return new RandomAccessBuffer(bytes);
                })
                .map(randomAccess -> {
                    try {
                        return new PDFParser(randomAccess).getPDDocument();
                    } catch (IOException e) {
                        throw new RuntimeException(e);
                    }
                });